tests/conftest.py

```diff
@@ -1,42 +1,142 @@
 # SPDX-License-Identifier: MIT
 
+import os
+import os.path
+import shutil
+import stat
+import sys
+import tarfile
+import tempfile
+
+import filelock
 import pytest
 
+import build
+
+
+if sys.version_info[0] == 2:
+    from urllib2 import urlopen
+else:
+    from urllib.request import urlopen
+
+
+INTEGRATION_SOURCES = {
+    'dateutil': ('dateutil/dateutil', '2.8.1'),
+    'pip': ('pypa/pip', '20.2.1'),
+    'Solaar': ('pwr-Solaar/Solaar', '1.0.3'),
+}
+
+
+def pytest_addoption(parser):
+    parser.addoption('--run-integration', action='store_true', help='run the integration tests')
+    parser.addoption('--only-integration', action='store_true', help='only run the integration tests')
+
+
+def pytest_collection_modifyitems(config, items):
+    skip_int = pytest.mark.skip(reason='integration tests not run')
+    skip_other = pytest.mark.skip(reason='only integration tests are run')
+
+    if config.getoption('--run-integration') and config.getoption('--only-integration'):  # pragma: no cover
+        raise pytest.UsageError("--run-integration and --only-integration can't be used together, choose one")
+
+    for item in items:
+        if os.path.basename(item.location[0]) == 'test_integration.py':  # pragma: no cover
+            if not config.getoption('--run-integration') and not config.getoption('--only-integration'):
```

Comment:
Feels this can no longer happen so you can drop the second part, no?

Reply:
No, I still have to check both, both flags enable running the integration tests. I can only skip the tests if both are false.

Reply:
Oh, I'm not in love with having two flags for this, IMHO you're widening and complicating the UI needlessly 🤷